test/performance/odp_sched_perf.c

```diff
@@ -550,10 +593,17 @@ static int create_pool(test_global_t *global)
 		max_num = pool_capa.buf.max_num;
 		max_size = pool_capa.buf.max_size;
 		max_uarea = pool_capa.buf.max_uarea_size;
-	} else {
+		max_pools = pool_capa.event_vector.max_pools;
+	} else if (test_options->pool_type == ODP_POOL_PACKET) {
 		max_num = pool_capa.pkt.max_num;
 		max_size = pool_capa.pkt.max_seg_len;
 		max_uarea = pool_capa.pkt.max_uarea_size;
+		max_pools = pool_capa.event_vector.max_pools;
+	} else {
+		max_num = pool_capa.event_vector.max_num;
+		max_size = pool_capa.event_vector.max_size;
+		max_uarea = pool_capa.event_vector.max_uarea_size;
+		max_pools = pool_capa.event_vector.max_pools;
```

Comment:
The capabilities should be checked for each pool type. In case of event vectors, buffer pool capabilities are currently not checked. One solution could be to move the checks to `create_pool_helper()`.